content/en/blog/2025/ux-research-prometheus-otel/index.md

```diff
@@ -0,0 +1,147 @@
+
+Regarding mental models, many users (both interviewees and survey respondents) don’t distinguish between resource attributes and Prometheus labels. They tend to think of them as the same thing.
+
+> "I would expect resource attributes as a rule to be treated exactly the same way as the attributes attached to the tracer, the metric... I wouldn't draw a boundary between them."  
+> — *Interview Participant 1*
+
+I also learned about the various workarounds people use to handle resource attribute issues in their specific use cases. Some promote selected resource attributes to labels, others handle the conversion at the OpenTelemetry level to avoid dealing with it in Prometheus, and a few convert all attributes, though usually only when the number of attributes is small.
+
+## Survey Insights
+
+The survey helped me quantify what I was hearing in interviews and reach a broader audience. At the time of writing, we’ve had 134 responses, with 61 from our target group—people who use OTel and Prometheus together.
+
+Here are the key findings:
+
+- Users don't conceptualize resource attributes as different from regular labels, yet current implementation treats them as separate metadata
+- Complex join syntax is a big barrier to adoption, with average developers unable to write basic queries accessing resource attributes
+- Manual attribute promotion creates operational overhead that scales poorly with team size and complexity
+- 78% of respondents find documentation gaps a challenge in their use of resource attributes
+
+![A bar chart showing challenges with OpenTelemetry resource attributes in Prometheus](Chart.PNG)
+
+The survey patterns were consistent with what emerged from my qualitative research. For detailed results, see the [anonymized survey responses](https://github.com/prometheus-community/ux-research/blob/main/prom-otel-research/survey-results.csv)
+
+## What We Didn’t Expect to Learn (But Did)
+
+We began this research to understand user pain points with resource attribute handling, but uncovered some unexpected and important findings.
+
+One of the most surprising was the realization that [OpenTelemetry’s Resource Detection feature](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource#telescoping) allows users to selectively retain or drop resource attributes based on relevance via a conceptual pattern sometimes referred to as “telescoping.” Despite its potential, many users and even some members of the Prometheus community seem unaware of it. This lack of awareness may have contributed to the adoption of the "join" pattern, which has since proven to be problematic.
```

Comment:
```suggestion
One of the most surprising was the realization that [OpenTelemetry’s Resource Detection feature](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource#telescoping) allows users to selectively retain or drop resource attributes based on relevance using a conceptual pattern sometimes referred to as “telescoping.” Despite its potential, many users and even some members of the Prometheus community seem unaware of it. This lack of awareness may have contributed to the adoption of the "join" pattern, which has since proven to be problematic.
```